pkg/utils/k8s/k8s.go

```diff
@@ -61,6 +61,13 @@ func (l StringMap) ToMap() map[string]string {
 	return l
 }
 
+func (l StringMap) AddIfNotEmpty(k, v string) map[string]string {
```

Comment:
Should this return StringMap instead? 
```suggestion
func (l StringMap) AddIfNotEmpty(k, v string) StringMap {
```